### PR TITLE
feat: IT for metadata events

### PR DESCRIPTION
**Pull request by olenagerasimova**
part of #1291 

Implemented integration tests for artifacts metadata using maven and docker. Also, simplified proxy repository settings to use single storage instead of one storage per remote.

### CODE REVIEW

File: .wiki/Configuration-Metadata.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    threads_count: 2 # optional, default 1
+    interval_seconds: 3 # optional, default 1
+```
+
+The essential here is `artifacts_database` section, other fields are optional. If `sqlite_data_file_path` field is absent,
+database file will be created at the parent location (directory) of the main configuration file. Metadata gathering
+mechanism uses [quartz](http://www.quartz-scheduler.org/) scheduler to process artifacts metadata under the hood. Quartz
+can be [configured separately](http://www.quartz-scheduler.org/documentation/quartz-2.1.7/configuration/ConfigMain.html),
+by default it uses `org.quartz.simpl.SimpleThreadPool` with 10 threads. If `threads_count` is lager than thread pool size,
+threads amount is limited to thread pool size.
```
Comment (dgarus) [inline .wiki/Configuration-Metadata.md:19]:
```suggestion
threads amount is limited to the thread pool size.
```

File: .wiki/Configuration-Metadata.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```yaml
+meta:
+  artifacts_database:
+    sqlite_data_file_path: /var/artipie/artifacts.db
+    threads_count: 2 # optional, default 1
+    interval_seconds: 3 # optional, default 1
+```
+
+The essential here is `artifacts_database` section, other fields are optional. If `sqlite_data_file_path` field is absent,
+database file will be created at the parent location (directory) of the main configuration file. Metadata gathering
```
Comment (dgarus) [inline .wiki/Configuration-Metadata.md:15]:
```suggestion
a database file will be created at the parent location (directory) of the main configuration file. The metadata gathering
```

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks!
There are a few minor comments.

**Comment by olenagerasimova** [comment]
@dgarus thanks, applied all the suggestions and added database table name into wiki.

**Review by dgarus — APPROVED** [review]
Thanks!

**Pull request merged**